### ISSUE TITLE
Can't Restore Session / Auth Failed after second program excecution

### ISSUE THREAD

**Issue by Maverick1387** [sochix/TLSharp#498]
Hi!

I can't send messages or do something else, after I closed and reopened my app.
At the first time, with code-request and makeAuth, everything works well. But when I start my app the next time, it won't authenticate correct.
(getting exception AUTH_KEY_UNREGISTERED)
The session.dat is build and up to date. I tried to delete it.
I tried with and without passing store to the constructor.
Nothing works...

my code:
```

FileSessionStore store = new FileSessionStore();
                
this.client = new TelegramClient(00000, "aaaaaaaaaaaaaaaaaaaaaaaaaaaa", store, "session"); 

await this.client.ConnectAsync();
                
bool con = this.client.IsConnected;
bool auth = this.client.IsUserAuthorized();
bool reg = await this.client.IsPhoneRegisteredAsync(number);
```

con, auth and reg are true! (after restart)
but when I execute 
`TeleSharp.TL.Contacts.TLFound found = await this.client.SearchUserAsync("somebody");`
for example, I get the 'AUTH_KEY_UNREGISTERED' exception.

Please help me! It cant be so difficult...

**Comment by Maverick1387** [comment]
Unbelievable... solved it... using 
' var store = new FileSessionStore(); '
instead of 
' FileSessionStore store = new FileSessionStore(); '
 
....

**Comment by Rynkadink** [comment]
I'm about to try that just for... well, just because everything else doesn't work for me.

**Comment by i3130002** [comment]
@Rynkadink  Maybe this would help https://github.com/sochix/TLSharp/issues/587

**Comment by jonathan3698** [comment]
Seriously, I got the same thing. Same code from OP, except it doesn't work. Same diagnostics aswell (IsUserAuthorized = true, etc). I have been searching for hours. I go on Gitter, no answer from anyone since like september. Also, there's no documentation about this, and what are the cause of the error. Wtf? I mean to provide a library you must atleast provide that...

Seriously, I would pay for that kind of tool if atleast it was documented properly, really frustrating... I understand no one is getting paid here, and I am free to go take other tools, but I'm just stating my opinion because your library is awesome except this thing that is really important which does not take that much time to do. You guys probably spent more hours answering to people like me complaining than you would have if you had done a good documentation...

And yes, I have read the telegram API documentation before saying this...

**Comment by knocte** [comment]
> You guys probably spent more hours answering to people like me complaining than you would have

What guys? Not sure who you're talking about, I don't think any of the 3 maintainers of TLSharp have ever answered here.

**Comment by knocte** [comment]
BTW this issue seems like a duplicate of #624 , which aarani is trying to fix with PR #629 . @jonathan3698 try that patch and let us know if it works for you.
